src/sota_tools/authenticate_test.cc

```diff
@@ -98,12 +99,22 @@ TEST(authenticate, no_json_zip) {
 /* Extract credentials from a provided JSON file. */
 TEST(authenticate, good_json) {
   // Authenticates with the ATS portal to the SaaS instance.
+  // Outdated. Keep backward compatibility
   boost::filesystem::path filepath = "tests/sota_tools/auth_test_good.json";
   TreehubServer treehub;
   int r = authenticate("", ServerCredentials(filepath), treehub);
   EXPECT_EQ(0, r);
 }
 
+TEST(authenticate, good_json_v2) {
+  // Authenticates with new backend.
+  // Note: update auth_test_goog_v2.json after deploy on prod. current file uses HAT
+  boost::filesystem::path filepath = "tests/sota_tools/auth_test_goog_v2.json";
```

Comment:
Is it intentional that the comment and path here are `goog`, but the json is `goox` and the zip is `good`?

I suggest not supporting json directly but only the zip file. I also suggest creating the zip file with a script at build or test time instead of hardcoding a binary file into the repo. We try to avoid binaries as much as possible, and it makes things much easier to read, debug, and maintain to do it with a script. There are some examples of this already in the repo and it was one of our goals to do this everywhere.

Reply:
typo fixed